src/client.ts

```diff
@@ -4840,13 +4840,8 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             $eventId: event.getId()!,
         });
 
-        const supportsThreadRR = this.canSupport.get(Feature.ThreadUnreadNotifications) !== ServerSupport.Unsupported;
-        if (supportsThreadRR && !unthreaded) {
-            const isThread = !!event.threadRootId;
-            body = {
-                ...body,
-                thread_id: isThread ? event.threadRootId : MAIN_ROOM_TIMELINE,
-            };
+        if (this.supportsExperimentalThreads() && !unthreaded) {
+            body.thread_id = event.threadRootId ? event.threadRootId : MAIN_ROOM_TIMELINE;
```

Comment:
I'm not really sure this change is related?

Reply:
True, that is not 100% related. I could remove it for now

Reply:
It might be fine to keep, I just didn't really understand it. 😄